src/main/scala/swiss/dasch/domain/ProjectService.scala

```diff
@@ -78,8 +68,21 @@ final case class ProjectService(
       .map(_ / "zipped")
       .flatMap(targetFolder => ZipUtility.zipFolder(projectPath, targetFolder).map(Some(_)))
 
-  def deleteProject(shortcode: ProjectShortcode): IO[IOException, Unit] =
-    findProject(shortcode).tapSome { case Some(prj) => Files.deleteRecursive(prj) }.unit
+  def deleteProject(shortcode: ProjectShortcode): Task[Unit] =
+    findProject(shortcode).tapSome { case Some(folder) =>
+      Files.deleteRecursive(folder) *> projects.deleteByShortcode(shortcode)
+    }.unit
+
+  def addProjectToDb(shortcode: ProjectShortcode): Task[Option[Project]] =
+    findProject(shortcode).flatMap {
```

Comment:
FYI: ZIO.foreach(_) { _ =>
 ...
}

Reply:
https://github.com/dasch-swiss/dsp-ingest/pull/232/commits/ef740e0aada821dbbca95ca8bd7a6f5ee596b266

Reply:
Amazing.

Reply:
Turns out that there is a even better way to write it by zipping both preconditions of the actual add to repo call:
```sclaa
 def addProjectToDb(shortcode: ProjectShortcode): Task[Unit] =
    (findProject(shortcode) <&> projects.findByShortcode(shortcode)).tapSome { case (Some(folder), None) =>
      projects.addProject(folder.shortcode).flatMap(p => ZIO.logInfo(s"Imported $folder as $p to database."))
    }.unit
```